pkg/constants/constants.go

```diff
@@ -430,6 +430,18 @@ const (
 	// installing. Use with caution.
 	ResumeSkipsClusterOperatorsLabel = "hive.openshift.io/resume-skips-cluster-operators"
 
+	// 	STSClusterLabel is an optional label that can be applied to ClusterDeployments to denote it uses short-lived
+	//	credentials. It is used as a label for some metrics.
+	STSClusterLabel = "api.openshift.com/sts"
+
+	// 	PrivateLinkClusterLabel is an optional label that can be applied to ClusterDeployments to denote it is uses
+	//	private-link. It is used as a label for some metrics.
+	PrivateLinkClusterLabel = "api.openshift.com/private-link"
+
+	// 	ManagedVPCLabel is an optional label that can be applied to ClusterDeployments to denote the VPC is managed by
+	//	SD. It is used as a label for some metrics.
+	ManagedVPCLabel = "api.openshift.com/managed-vpc"
+
```

Comment:
This is tickling my proprietary-ness antenna. As a rule, it's not a great look to add code that's explicitly tied to a particular consumer. Of course we add stuff at the request of consumers and for specific consumer use cases, but we usually try to design/position it so it's generically useful, at least in theory.

Ideally, we would find some way to discover and report these metric labels without using the SD-specific CD labels. E.g. we can certainly tell whether we're privatelink based on `spec.platform.aws.privateLink != nil`. Not sure about the others.